### ISSUE TITLE
Enclosure Question

### ISSUE THREAD

**Issue by ssmoss** [ssmoss/homebridge-flume#1]
@notms I saw you posted on a question on the `genmon` repo. I didn't want to clutter the issue tracker system.  I was curious do you have a 22kW generator and what type of enclosure do you use? I am trying to figure out if the one I have is too large to fit inside the generator. Thanks :-)

**Comment by ssmoss** [comment]
Did you use a drill to create a hole on the bottom to route the data + power cables?

**Comment by notms** [comment]
Yes a 22 and this is the enclosure DJC Supply Co. 3.9" x 5.9" x 2.75" ABS Waterproof IP67 Hinged Transparent Clear Cover Electronics Box with Mounting Plate, Bracket https://a.co/d/0B9ntXY I did use a drill at the bottom and used a weatherproof wiring guide

**Comment by ssmoss** [comment]
Thanks so much for getting back to me! I was looking at this [enclosure](https://www.amazon.com/dp/B097XQVMY6) (7.9”×6.1”×3.1”) which seems a bit bigger than yours. Do you have a rough idea how much working space there is?

For the weatherproof guide, how did you contend with the data cable having 8 separate wires? While it could be fed though would that 8 separates cables not have a good seal?

**Comment by notms** [comment]
I used this, it provides an amazing weather seal and squeezes all the cables together TUPARKA 50 Pack Cable Gland Waterproof Adjustable 3-16mm Cable Connectors PG7 PG9 PG11 PG13.5 PG16 PG19 Plastic Cable Gland Joints with Gaskets in PP Box https://a.co/d/fvXytE6

**Comment by notms** [comment]
What I like about mine is I can quickly get into the box as well as clearly see all the lights blinking even from outside through the slits of the generator.

**Comment by ssmoss** [comment]
Yea, I am not heart set on the one I got, I wanted to put both the Pi and the Splitter in the box so I am little concerned that 3.9" x 5.9" x 2.75" would be too small. I ordered one along with the next size up 6.7" x 8.7" x 4.3" to see. I was looking at your pictures from issue 940 which I think are the best I have seen of that area of the generator. I think the extra height and depth would be okay, but I am concerned about going from 3.9 to 6.7 on the width.

**Comment by notms** [comment]
From the pictures when I put my box in there was still some room above as well as at the bottom and to the right. Just to let you know I can fit two raspberry pi zeros as well as two converters easily.

**Comment by ssmoss** [comment]
So noted. Thank you so much!

For my setup I went with the PI 3A+ and the Splitter from PintSize.me.

**Comment by notms** [comment]
This is the splitter that I went with and a cable 3pcs TTL to RS232 Module TTL... https://www.amazon.com/dp/B091TN2ZPY?ref=ppx_pop_mob_ap_share and StarTech.com EPS 8 Pin Power... https://www.amazon.com/dp/B000M802RG?ref=ppx_pop_mob_ap_share
The smallest setup that I could find with lights that tell me the exact status plus gives me extra parts

**Comment by ssmoss** [comment]
Very cool. You have given me a lot to think about. I am going to lay the stuff out and see. I'll report back if you are interested. Thank you so much all the input. I hope they can sort out the issue with your wifi controller. The two maintainers are crazy helpful. This is such an amazing project.

**Comment by notms** [comment]
Definitely, just remember if you are doing wifi, the generac enclosure is like a faraday cage, my eero router is literrraly 25 inches from the generator and I get about -70db on internal wifi, if you have external antenna I would suggest to use if you want wifi.

**Comment by ssmoss** [comment]
Yea, the plan is to do wifi. Does the Zero have an external wifi antenna? I know the 3A+ doesn't. I wonder if I should get a USB WIFI module with an antenna?

**Comment by notms** [comment]
Nope zero doesn’t, there is a way I can break the connection and splice a connector but a bit risky, some PI’s do have external antennas, I was thinking of using the usb wifi, some  are good and some are junk

**Comment by ssmoss** [comment]
Yea my AP would be inside the house, maybe 10 - 15 feet Birds Eye. If yours is only 25 inches away that makes me really concerned.

I found this https://www.amazon.com/TP-Link-TL-WN722N-Wireless-network-Adapter/dp/B002SZEOLG and TP Link normally makes good stuff.

**Comment by ssmoss** [comment]
I just ordered a couple of USB ones with removal antenna. I will report back if any worked out of the box.

**Comment by ssmoss** [comment]
@notms Just an FYI I got an USB Wifi module working that has a removal antenna. I am going to have this be the primary interface for the Pi and the onboard as a fallback. See https://github.com/jgyates/genmon/issues/670#issuecomment-1716728015 for more details.

**Comment by notms** [comment]
Very cool, can you share the link as well as any config as well as extra power that the external USB is taking, I am running Zero so need to keep it all in mind. I am assuming you would want to buy an external antenna that mounts to outside with weather proof seal as well.

**Comment by ssmoss** [comment]
Sure thing, so I am using this [hub](https://www.amazon.com/dp/B01N6MF654), this [adaptor](https://www.amazon.com/dp/B01CCMUN8C), this [antenna extension cable](https://www.amazon.com/dp/B0C7HJS3QW), these [usb 180 / 90 degree angled adaptors](https://www.amazon.com/dp/B0C7QSXCY3) and [usb 90 degree angled adaptors](https://www.amazon.com/dp/B0BZ7V775L), and possibly this [antenna](https://www.amazon.com/dp/B094MQ9X7N) or the plastic one that came with the adaptor.

To get the adaptor working I had to use [RTL8811CU_for_Raspbian](https://github.com/fastoe/RTL8811CU_for_Raspbian) and depending on if you are using `32 bit` or `64 bit` you might run into this [issue](https://github.com/fastoe/RTL8811CU_for_Raspbian/issues/11) (TLDR; use `make ARCH="arm64”` when using a `64 bit` OS).

No other settings are needed and both adaptor will connect and use the same wireless network. I set the USB one as the primary interface and use the on-board one as the fall back.

I just got all everything put inside the enclosure (Pi + HAT + Splitter) and I can add a picture or two if you are interested.

**Comment by notms** [comment]
That is very cool thank you, if you can share a picture I would appreciate it. I was looking to find a USB adapter, I will go with yours just ordered for the antenna https://a.co/d/cZvW8PU and https://a.co/d/iUScPuT to make sure it’s weather proof and has enough length to get to the enclosure. How far is your router from your antenna and also have you tested the gain range that you get? I also found a way to get to the raspberry pi externally without opening up any ports on the router if you are interested I can share.

**Comment by ssmoss** [comment]
Yea I figured knowing my luck I would need to connect a keyboard and mouse to troubleshoot and I didn't want to lose wireless. I picked that one because the 3 has a full-size USB. I am not sure if the antenna that I got is "weather" proof though I wasn't too concerned, I figured there isn't any power running over it so I didn't think it would matter to much?

It's going to be like ~10 feet as the crow flys I think. Though it will be going though a wall. I know standing in the spot the generator will go I have decent wireless signal on my phone so I think it is close enough?

Sure, I would be happy to see what you did!

I am not sure if you read my longer issue on the `genmon` page but I have a home automation PC that I am using a proxy for the Pi via a SSH tunnel. My `genmon` runs on local host so it isn't inaccessible without the use of the SSH tunnel. I saw that one person that got hacked and I put my tiny foil hat on. Being able to get to externally would be cool but I wasn't quite sure the best way to do that.

**Comment by notms** [comment]
I got a hold of Tailscale VPN solution at these requirements it’s free and allows access without opening any ports or proxies. in Genmon can you share what your connection quality is with or without the antennae as my router is about 3 feet from the generator.

**Comment by ssmoss** [comment]
Oh nice, I am going to look into them now.

I can once I get the generator installed. I don't have a way to power the Pi without it.

**Comment by notms** [comment]
I also got a hold of a UPS cyber power 1500 that will sustain the cable modem and the router so that I can see the status of the generator even if it’s not providing power and there is no external power, it’s limited but should run a couple of hours.

**Comment by ssmoss** [comment]
Yea that should be super useful during the transition when the power goes out. My understanding is it takes like ~30 seconds to the generator to start so you need a battery backup to sustain things for like 5 minutes.

We have some weird voltage issues in the area so all the computers have a battery back up to do voltage smoothing. Thankfully the network gear is by a computer so they share the unit.

Thanks for reminding me, I have to pick up a small one for the access point the generator connects to.

**Comment by ssmoss** [comment]
So you run Tailscale on your home network and then VPN in when external to access it?

**Comment by notms** [comment]
It runs right on the PI as a service, it’s very cool

**Comment by ssmoss** [comment]
Ah so you can access the Pi on your local network and then the client adds to the virtual Tailscale network that you can VPN in so it doesn't expose your entire network?

That's pretty cool.

**Comment by notms** [comment]
Yes essentially you download a Tailscale vpn app on your phone and once
enabled the external IP pops up to use

**Comment by notms** [comment]
I went through all the steps and can’t see the Realtek usb wifi adapter. I can see it when I go to lsusb but don’t see it if I do ifconfig curious if you ran into it before, thanks

**Comment by ssmoss** [comment]
What is the output of `lsusb`? If you look at the bottom of the adaptor does it list the chipset? On the bottom of mine it had it printed. Can you post that as well. Can you post all the steps you did to `build` and `install` the driver?

**Comment by notms** [comment]
Thanks for your reply I re-ran the install rebooted multiple times and finally got it working. How did you prioritize wifi decide vs wifi connection?

**Comment by ssmoss** [comment]
Make sure it survives a couple of reboots just to make sure :-)

So each adaptor will have a different MAC address to identify it. In my router I assigned a static IP to one. To use the USB one I connect to that IP.

**Comment by notms** [comment]
So you mapped two MAC addresses from two different wifi connections to a single IP?

**Comment by ssmoss** [comment]
No, I have the internal one with the MAC address of `AA:AA:AA:AA:AA:AA` and the usb one with the MAC address of `BB:BB:BB:BB:BB:BB`. In my router I said map `10.0.0.1` -> `AA:AA:AA:AA:AA:AA` and `10.0.0.2` -> `BB:BB:BB:BB:BB:BB`. To access on the USB one I would use address `10.0.0.2`

**Comment by ssmoss** [comment]
The author of the project added two configuration settings because of a ticket I made. They are:

```
flask_listen_ip_address=
genmon_server_address=
```

If you want to bind `genmon` to a specific interface you can change it from the default of `0.0.0.0` which is listen on all interfaces.

**Comment by notms** [comment]
Ahh ok that makes sense so you have them both running so if one fails you would hope the other is able to communicate out, but it doesn’t matter which one is receiving or sending data from your perspective

**Comment by ssmoss** [comment]
Yes so since it listening on `0.0.0.0` you should be able to load the UI from either address.

In my case, no, because I didn't SSL certificate rotation (restarting), wanting better dynamic user accounts, and wanting to lock the PI down. As such my `genmon` listens on `127.0.0.1` and I create a SSL tunnel to it. I then use the Apache instances running on my home automation computer to handle SSL, user access control and as a proxy via the tunnel to `genmon`.

Once I everything into place and see what the wireless single is like for the internal adaptor I might create a second tunnel and then use Apache to create a status based load balances to automatically change if one goes down.

Given that a person that got their instance hacked because they exposed it publicly I am being super careful. Also since the PI is outside I wanted to limit what it has access to in the very rare chance of theft.

**Comment by ssmoss** [comment]
Once you get your two IP address setup, let me know if you have any issues connecting.

**Comment by notms** [comment]
I got two IP’s running just need to tie it to the USB with external antenna as a priority and then internal wifi as backup if the first fails. It looks like it can be done with SSID’s but can’t find to do with wifi hardware devices.

**Comment by ssmoss** [comment]
Not sure I am following? Out of the box both adaptor should just connect to your wireless network. If you run `iwconfig` you see what each one is doing. You can use `ifconfig` to get the MAC address for doing the static address mapping. Look for the section titled `ether` in the output of `wlan0` and `wlan1`.

How is [this](https://www.amazon.com/dp/B095JTSQ58) antenna working for you? I was thinking going with it instead and then using magnets instead of screws to stick to the outside of the enclosure

**Comment by notms** [comment]
The taller antenna is actually giving me 10 dBm worse at 5Ghz, trying to get this to 2.4Ghz to see if I see better results.

**Comment by ssmoss** [comment]
I wouldn't bother with 5GHz as it drops off quickly with obstructions. I will be using 2.4GHz for mine. I don't suspect you need a super high speed link for `genmon`, I think something around 10Mb/s would be more than enough.

**Comment by notms** [comment]
I am trying to get the usb adapter to connect to 2.4, I have Google nest wifi, no easy way to force 2.4 unless I can disable 5ghz in driver settings somewhere

**Comment by ssmoss** [comment]
Ah...hum...I think you can set a frequency list in `wpa_supplicant.conf`

**Comment by notms** [comment]
Oh cool will check it out btw I found command “is dev” pretty useful as it calls out the channel width and center as well as to power to .00 decimal levels

**Comment by ssmoss** [comment]
According to this [wpa_supplicant.conf](https://web.mit.edu/freebsd/head/contrib/wpa/wpa_supplicant/wpa_supplicant.conf) it seems like you could set `freq_list`:

```
# freq_list: Array of allowed frequencies
# Space-separated list of frequencies in MHz to allow for selecting the BSS. If
# set, scan results that do not match any of the specified frequencies are not
# considered when selecting a BSS.
```

I have seen some places indicate that this option may or may not work so it may take some testing. Are you have to make a secondary wireless network that only runs on 2.4 GHz?

**Comment by notms** [comment]
I got it working with forcing 2.4 unfortunately no change on dBm with the tap antenna at least I am not loosing signal as I have been with 5ghz. I am doing this test outside of the generator just to get a better understanding, I will report back in a couple of weeks when I install it on the generator. I am getting the hockey puck antenna tomorrow will test to see how it responds and report back. Thanks for your help btw really appreciate it

**Comment by notms** [comment]
I am also using a raspberry zero so power consumption could be an issue, I will try with a USB powered hub to see if I get better numbers as well

**Comment by ssmoss** [comment]
Yay, happy you got that all sorted out!

Oh interesting point. I think you can check the power going to the device using `lsusb -v|egrep "^Bus|MaxPower"`. Mine shows 500mA which is the max permissible over USB 2 I think. I would think if you are using supported adaptor or the power from the generator battery you should be okay? I am curious to know your results.

**Comment by notms** [comment]
Thank you showing 500mA

**Comment by ssmoss** [comment]
So I think you are okay power wise. Though given that it is USB never hurts to double check :-) Let me know what you find out.

**Comment by notms** [comment]
So it’s interesting the two external antennas dropped the connection worse by like 10 dbm if not more, not sure if I need an amp and this is just in my home setting not at the generator. I will hopefully test this next weekend with the generator to see if what real world numbers would be.

**Comment by ssmoss** [comment]
So I got the [wireless antenna](https://www.amazon.com/dp/B095JTSQ58) you suggested connected and it reports better compared to the internal wireless. The internal reports the following:
```
Link Quality=50/70  Signal level=-60 dBm
Bit Rate=39 Mb/s
```
where as the USB one reports:
```
Link Quality=66/100  Signal level=31/100  Noise level=0/100
Bit Rate:72.2 Mb/s
```
I currently have my box upside-down on the table letting the glue for the magnets dry.

**Comment by notms** [comment]
That looks pretty good glad you are seeing the difference, I will be doing more tests, what I like is that it’s weather proof which is key in longevity

**Comment by ssmoss** [comment]
@notms I noticed something after I started doing some practical testing with my enclosure. You need to adjust the route metric for the `usb` adaptor due to which interface it favors. Networking routes are prioritized based on the following:
```
Routes are prioritized by Metrics. The lower the metric the higher the precedence.
Once we have the routes created with Metrics specified, Route with Lower Metric
value will take higher Precedence.
```
Here is the current routing table:
```
# route
Kernel IP routing table
Destination        Gateway            Genmask           Flags Metric Ref    Use Iface
default            xxx.xxx.xxx.xxx    0.0.0.0           UG    302    0        0 wlan0 <-- onboard
default            xxx.xxx.xxx.xxx    0.0.0.0           UG    306    0        0 wlan1 <-- wifi
xxx.xxx.xxx.xxx    0.0.0.0            xxx.xxx.xxx.xxx   U     302    0        0 wlan0
xxx.xxx.xxx.xxx    0.0.0.0            xxx.xxx.xxx.xxx   U     306    0        0 wlan1
```
You will see that the `onboard` adaptor has a lower `metric` compared to the `usb` one which means that it will be favored. To fix this you need to edit the `/etc/dhcpcd.conf ` file and add the following:
```
# Favor USB Wifi Of Internal
interface wlan1
metric 100
```
Either reboot or run the following command `service dhcpcd restart`. If you check the routing table after this you will see the following:
```
# route
Kernel IP routing table
Destination        Gateway            Genmask           Flags Metric Ref    Use Iface
default            xxx.xxx.xxx.xxx    0.0.0.0           UG    100    0        0 wlan1 <-- onboard
default            xxx.xxx.xxx.xxx    0.0.0.0           UG    302    0        0 wlan0 <-- wifi
xxx.xxx.xxx.xxx    0.0.0.0            xxx.xxx.xxx.xxx   U     100    0        0 wlan1
xxx.xxx.xxx.xxx    0.0.0.0            xxx.xxx.xxx.xxx   U     302    0        0 wlan0
```

**Comment by ssmoss** [comment]
@notms So I was able to actually test my box today and the wireless usb adaptor doesn't work. When the generator kicks on something happens and it just disappears. As a result I pulled it from the box and I am using the onboard one. The signal strength seems passable:
```
Link Quality=35/70  Signal level=-75 dBm
Bit Rate=26 Mb/s   Tx-Power=31 dBm
```
It isn't great but 26Mb/s should be more than enough.

**Comment by notms** [comment]
Ohh interesting, you loosing the raspberry pi? Or only adapter? EMP possibly interfering? Thank you for sharing, I am yet to install it, possibly no power for PI to power the adapter?

**Comment by ssmoss** [comment]
The Pi seems fine though my SSH sessions became unresponsive. It would eventually allow me to reconnect once the routing updated. I am powering the Pi from the battery so in theory it shouldn't be an issue unless the generator turning over causes a dip. I am not 100% sure that is it is a power issue, its possible it is some time of interference. The built in one seems to be functional okay so that is good at least. I am going to monitor some more and see.

**Comment by ssmoss** [comment]
So I found [this](https://github.com/jgyates/genmon/wiki/Appendix-M-Tips-and-Tricks-useful-for-Genmon#configuring-usb-wifi-on-a-raspberry-pi) section after digging though the [genmon](https://github.com/jgyates/genmon). It was reported that [Mideatek RT5370N Wireless-N USB Adapter from Amazon](https://www.amazon.com/gp/product/B073J3HXZH) had worked but of course it is no longer in stock. That said the important part is the `RT5370N` which is the chipset used by the USB module. I also found a list of supposedly [supported](https://elinux.org/RPi_USB_Wi-Fi_Adapters) wireless adapters. This confirmed that chipset is supported by Raspbian. I found [AR9271 802.11n 150Mbps Wireless USB WiFi Adapter Network ](https://www.amazon.com/dp/B07FVRKCZJ) which according to the support wireless adapter link works if you install the `aircrack-ng` package. I finally took to eBay to see if there were any other options and I found [150M USB WiFi Wireless Adapter LAN w/Antenna Raspberry Pi2 B+ralink rt5370Chi-^C](https://www.ebay.com/itm/355040854071). Given how cheap they are and the fact I could return the ones from Amazon I picked one of these up as well. Just make sure the one you get has a removal antenna. I almost got one that didn't.

I am going to return the hub and connect directly as now that I know the internal adapter works I am less concerned about needing to have a free USB port.

For reference these are some of the errors I saw in dmesg:
```
[ 1230.468383] usb usb1-port1: disabled by hub (EMI?), re-enabling...
[ 1230.468420] usb 1-1: USB disconnect, device number 2
[ 1230.533001] rtl8821cu 1-1:1.0: Runtime PM usage count underflow!
[ 1230.680364] Indeed it is in host mode hprt0 = 00001501
[ 1230.872320] usb 1-1: new high-speed USB device number 3 using dwc_otg
[ 1230.872506] Indeed it is in host mode hprt0 = 00001101
[ 1231.060408] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1231.060457] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1231.060481] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1231.268353] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1231.268404] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1231.268425] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1231.480322] usb 1-1: device not accepting address 3, error -71
[ 1231.480498] Indeed it is in host mode hprt0 = 00001501
[ 1231.672287] usb 1-1: new high-speed USB device number 4 using dwc_otg
[ 1231.672435] Indeed it is in host mode hprt0 = 00001101
[ 1231.880932] usb 1-1: New USB device found, idVendor=0bda, idProduct=c811, bcdDevice= 2.00
[ 1231.880962] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 1231.880973] usb 1-1: Product: 802.11ac NIC
[ 1231.880982] usb 1-1: Manufacturer: Realtek
[ 1231.880991] usb 1-1: SerialNumber: 123456
[ 1234.436631] usb usb1-port1: disabled by hub (EMI?), re-enabling...
[ 1234.436690] usb 1-1: USB disconnect, device number 4
[ 1234.580565] Indeed it is in host mode hprt0 = 00001501
[ 1234.768491] usb 1-1: new high-speed USB device number 5 using dwc_otg
[ 1234.768730] Indeed it is in host mode hprt0 = 00001101
[ 1234.977327] usb 1-1: New USB device found, idVendor=0bda, idProduct=c811, bcdDevice= 2.00
[ 1234.977368] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 1234.977387] usb 1-1: Product: 802.11ac NIC
[ 1234.977403] usb 1-1: Manufacturer: Realtek
[ 1234.977419] usb 1-1: SerialNumber: 123456
[ 1236.377397] usb usb1-port1: disabled by hub (EMI?), re-enabling...
[ 1236.377444] usb 1-1: USB disconnect, device number 5
[ 1236.524493] Indeed it is in host mode hprt0 = 00001501
[ 1236.712415] usb 1-1: new high-speed USB device number 6 using dwc_otg
[ 1236.712561] Indeed it is in host mode hprt0 = 00001101
[ 1236.928675] usb 1-1: device descriptor read/all, error -71
[ 1236.928813] Indeed it is in host mode hprt0 = 00001101
[ 1237.116441] usb 1-1: new high-speed USB device number 7 using dwc_otg
[ 1237.116601] Indeed it is in host mode hprt0 = 00001101
[ 1237.325041] usb 1-1: New USB device found, idVendor=0bda, idProduct=c811, bcdDevice= 2.00
[ 1237.325069] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 1237.325080] usb 1-1: Product: 802.11ac NIC
[ 1237.325089] usb 1-1: Manufacturer: Realtek
[ 1237.325098] usb 1-1: SerialNumber: 123456
[ 1238.602122] usb usb1-port1: disabled by hub (EMI?), re-enabling...
[ 1238.602155] usb 1-1: USB disconnect, device number 7
[ 1238.744530] Indeed it is in host mode hprt0 = 00001501
[ 1238.932484] usb 1-1: new high-speed USB device number 8 using dwc_otg
[ 1238.932642] Indeed it is in host mode hprt0 = 00001101
[ 1239.141066] usb 1-1: New USB device found, idVendor=0bda, idProduct=c811, bcdDevice= 2.00
[ 1239.141095] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 1239.141106] usb 1-1: Product: 802.11ac NIC
[ 1239.141115] usb 1-1: Manufacturer: Realtek
[ 1239.141124] usb 1-1: SerialNumber: 123456
[ 1240.350437] usb usb1-port1: disabled by hub (EMI?), re-enabling...
[ 1240.350472] usb 1-1: USB disconnect, device number 8
[ 1240.492646] Indeed it is in host mode hprt0 = 00001501
[ 1240.624834] Indeed it is in host mode hprt0 = 00001501
[ 1240.952523] usb 1-1: new high-speed USB device number 9 using dwc_otg
[ 1240.952849] Indeed it is in host mode hprt0 = 00001101
[ 1241.140511] usb 1-1: device descriptor read/64, error -71
[ 1241.248654] Indeed it is in host mode hprt0 = 00001101
[ 1241.456550] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1241.456587] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1241.456605] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1241.457356] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1241.457387] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1241.457413] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1241.457800] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1241.457825] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1241.457847] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1241.458256] usb 1-1: device descriptor read/all, error -71
[ 1241.458358] Indeed it is in host mode hprt0 = 00001501
[ 1241.584583] Indeed it is in host mode hprt0 = 00001501
[ 1241.912545] usb 1-1: new high-speed USB device number 10 using dwc_otg
[ 1241.912693] Indeed it is in host mode hprt0 = 00001101
[ 1242.121091] usb 1-1: New USB device found, idVendor=0bda, idProduct=c811, bcdDevice= 2.00
[ 1242.121119] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 1242.121130] usb 1-1: Product: 802.11ac NIC
[ 1242.121139] usb 1-1: Manufacturer: Realtek
[ 1242.121148] usb 1-1: SerialNumber: 123456
[ 1243.284742] usb usb1-port1: disabled by hub (EMI?), re-enabling...
[ 1243.284773] usb 1-1: USB disconnect, device number 10
[ 1243.428639] Indeed it is in host mode hprt0 = 00001501
[ 1243.620682] usb 1-1: new high-speed USB device number 11 using dwc_otg
[ 1243.621094] Indeed it is in host mode hprt0 = 00001101
[ 1243.808590] usb 1-1: device descriptor read/64, error -71
[ 1243.916718] Indeed it is in host mode hprt0 = 00001101
[ 1244.044642] Indeed it is in host mode hprt0 = 00001501
[ 1244.393448] usb 1-1: New USB device found, idVendor=0bda, idProduct=c811, bcdDevice= 2.00
[ 1244.393477] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 1244.393488] usb 1-1: Product: 802.11ac NIC
[ 1244.393497] usb 1-1: Manufacturer: Realtek
[ 1244.393506] usb 1-1: SerialNumber: 123456
[ 1245.589320] usb usb1-port1: disabled by hub (EMI?), re-enabling...
[ 1245.589353] usb 1-1: USB disconnect, device number 11
[ 1245.732710] Indeed it is in host mode hprt0 = 00001501
[ 1245.920634] usb 1-1: new high-speed USB device number 12 using dwc_otg
[ 1245.920801] Indeed it is in host mode hprt0 = 00001101
[ 1246.129227] usb 1-1: New USB device found, idVendor=0bda, idProduct=c811, bcdDevice= 2.00
[ 1246.129255] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 1246.129266] usb 1-1: Product: 802.11ac NIC
[ 1246.129275] usb 1-1: Manufacturer: Realtek
[ 1246.129284] usb 1-1: SerialNumber: 123456
[ 1247.314134] usb usb1-port1: disabled by hub (EMI?), re-enabling...
[ 1247.314166] usb 1-1: USB disconnect, device number 12
[ 1247.456758] Indeed it is in host mode hprt0 = 00001501
[ 1247.644674] usb 1-1: new high-speed USB device number 13 using dwc_otg
[ 1247.644828] Indeed it is in host mode hprt0 = 00001101
[ 1247.853341] usb 1-1: New USB device found, idVendor=0bda, idProduct=c811, bcdDevice= 2.00
[ 1247.853381] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 1247.853392] usb 1-1: Product: 802.11ac NIC
[ 1247.853401] usb 1-1: Manufacturer: Realtek
[ 1247.853410] usb 1-1: SerialNumber: 123456
[ 1249.023536] usb usb1-port1: disabled by hub (EMI?), re-enabling...
[ 1249.023570] usb 1-1: USB disconnect, device number 13
[ 1249.164778] Indeed it is in host mode hprt0 = 00001501
[ 1249.352775] usb 1-1: new high-speed USB device number 14 using dwc_otg
[ 1249.352934] Indeed it is in host mode hprt0 = 00001101
[ 1249.560993] usb 1-1: device descriptor read/all, error -71
[ 1249.561118] Indeed it is in host mode hprt0 = 00001101
[ 1249.748716] usb 1-1: new high-speed USB device number 15 using dwc_otg
[ 1249.748865] Indeed it is in host mode hprt0 = 00001101
[ 1249.936771] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1249.936809] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1249.936827] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1250.144836] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1250.144886] NYET/NAK/ACK/other in non-error case, 0x00000002
[ 1250.144914] NYET/NAK/ACK/other in non-error case, 0x00000002
```

**Comment by ssmoss** [comment]
So I finally got the the USB wireless module installed. I used [this](https://www.amazon.com/dp/B07CF7243G) to give me more flexibility with this [this](https://www.amazon.com/dp/B086D72XR6) adapter. It was recognized as soon as I plugged it in. It wouldn't connect the network without an antenna connected but once it did the reception is great.

You can see the chart before 15:45 the low quality of the internal, how much it gets boosted when the side comes off, and finally how it returns to the low quality once the side goes back on. This is in comparison to usb one with the external antenna attached.

![Wireless Strength](https://github.com/ssmoss/homebridge-flume/assets/8904909/ed2ef035-a266-4586-952c-8feea17ba762)

**Comment by ssmoss** [comment]
@notms so more updates...man this has been quite the learning experience. The adaptor I mentioned in my last post is a no go. The built-in drivers only work for a limited time and it has some strange disconnection / reconnection issue. I went back to [EDUP AC600M](https://www.amazon.com/gp/product/B01CCMUN8C) which is not supported out of the box. I used the drivers from [morrownr](https://github.com/morrownr/8821cu-20210916) instead. They are much more recent and support most of the now common features for wireless. The drivers took awhile to compile, about ten minutes so I wouldn't do it if you are in a rush. I had issues building it via a remote session (ssh) so I ended up bring the Pi inside so I could use a monitor and keyboard and building directly from the console. So far things are working much much better. The logs output during connect / dhcp refresh seem more normal and it don't seem to exhibit the same strange disconnect issue. The reception also seems better as well. I will keep monitoring and report back but I think this might be the way to go.
